### ISSUE TITLE
(One#26752818) [Insert new step]: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies.

### ISSUE THREAD

**Issue by ek68794998** [Azure/LogicAppsUX#4292]
### **User Experience:**

Visually impaired users who depend on assistive technology as screen readers will face difficulty while navigating on nested interactive controls or can cause focus problems due to which user will not be able to perform tasks properly.

**Note: Login Credentials must not be provided within the bug.**

### **Pre-requisites:**

1.  First we need to create a flow to test this feature, a person can create their own flow

### **Repro Steps:**

1.  Open URL "<https://make.powerautomate.com/>" and sign in with Valid credentials.
2.  Flow Homepage will appear on the screen, navigate to 'My Flows' from the left pane and select it.
3.  My Flows Page will appear on the screen which consist a list of already created flows.
4.  Navigate to one of the flows on the list and click the icon for "Edit." You are now in the flow designer.

### **Actual Result:**

Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies.

**Observation:** Screen reader is remaining silent when focus moved to Insert a new step control in the flow design when navigated through tab control

### **Expected Result:**

Nested interactive control "tab control" should be defined properly i.e. "Tab control" must not have any focusable descendants.

### **Issue:**

Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (nested-interactive - <https://dequeuniversity.com/rules/axe/4.7/nested-interactive?application=msftAI>)

### **Target application:**

Microsoft Power Automate -  <https://make.preview.powerautomate.com/environments/839eace6-59ab-4243-97ec-a5b8fcc104e4/flows/3b6866d6-01d0-4505-a5c3-7d51330eaf7b?v3=true> 

### **Element path:**

.edge-drop-zone-container > .msla-drop-zone-viewmanager2 > div[data-tabster="{\"restorer\":{\"type\":1}}"][role="button"]

### **Snippet:**

```
<div tabindex="-1" role="button"
data-tabster="{&quot;restorer&quot;:{&quot;type&quot;:1}}"
aria-expanded="false">
```

### **Related paths:**

#msla-edge-button-send_a_push_notification-undefined

### **How to fix:**

**Fix any of the following:**

-   Element has focusable descendants

Environment: Microsoft Edge version 121.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at  <http://aka.ms/AccessibilityInsights>.

### **MAS Reference:**

[MAS 4.1.2 -- Name, Role, Value](https://aka.ms/MAS4.1.2)

### **ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:**

<https://www.w3.org/WAI/ARIA/apg/patterns/tabs/>

### **UIA/DOM details available:**

Yes

### **Reference Links:**

[Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.

External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [External Bug Process (sharepoint.com)](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDN8MDF8fDE1MThmM2JlZGFjMDRlZDhiZmViYTUxMDA2M2NiZWI3fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc3NDEyNTM2MTEzNzIwNzR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaUlpd2lRVTRpT2lJaUxDSlhWQ0k2TVRKOQ%3D%3D&sdata=MnlpSXlVMnpGT3cvZ0xJNnFWUWZWTGt6dmo1VkJMZFpIZHBvUUduNDhyMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-kgeetala%40microsoft.com). "HCL Staff should not log any third-party external bugs. Should be routed to EDAD team".

Please reach out to [C&AI Teams channel](https://apc01.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cAklavyaM%40hcl.com%7c24e89c6ac26f42cfe40c08d7667e49f3%7c189de737c93a4f5a8b686f4ca9941912%7c0%7c0%7c637090564906844600&sdata=22JjBMoA1ID/KnYqFa/OB%2BrkTaLh5YYsGBd/7Bm7u5w%3D&reserved=0) for any process-related queries.

### Are you using new designer or old designer

New Designer (With action panels)

### Screenshots or Videos

![MAS4 1 2 Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies](https://github.com/Azure/LogicAppsUX/assets/1350074/37891cf9-0506-45dd-9a0a-2cbb7b202149)

### Browser

Microsoft Edge version 121.0.0.0

### Additional context

[One#26752818](https://dev.azure.com/msazure/One/_workitems/edit/26752818)

**Comment by edgarguoms** [comment]
FYI, the ADO is tagged as A11ySev3.
